tests/test_sliders/test_labeled_slider.py

```diff
@@ -85,3 +85,43 @@ def test_editing_float(qtbot):
     slider._label.setValue(0.5)
     slider._label.editingFinished.emit()
     assert slider.value() == 0.5
+
+
+@pytest.mark.parametrize("cls", [QLabeledSlider, QLabeledDoubleSlider])
+def test_extended_label_spinbox_range(cls, qtbot):
+    slider = cls()
+    mock = Mock()
+    qtbot.addWidget(slider)
+    slider.setRange(0, 10)
+    slider.rangeChanged.connect(mock)
+    assert slider._label.minimum() == 0
+    assert slider._label.maximum() == 10
+    assert slider.minimum() == 0
+    assert slider.maximum() == 10
+
+    # changing slider value without changing spinbox range should return a clipped value
+    # slider._label.setValue(20)
+    # slider._label.editingFinished.emit()  # simulate editing the label manually
+    # qtbot.wait(20)
+    # assert slider.value() == 10
+    # assert slider.maximum() == 10
+
+    # after changing label range, setting the label to a value outside the range should
+    # update the slider min/max as appropriate
+    slider.setLabelSpinBoxRange(-10, 20)
+
+    with qtbot.waitSignal(slider.rangeChanged):
+        slider._label.setValue(15)
+        assert slider._label.minimum() == -10
+        slider._label.editingFinished.emit()  # simulate editing the label manually
+        assert slider._label.minimum() == -10
```

Comment:
something is going wrong here

`editingFinished` is the signal used to propagate label value changes down to the slider so I manually emit it for the test. This has the unintended side effect of changing `slider._label.minimum()`.

I tried to fix this by modifying 
https://github.com/pyapp-kit/superqt/blob/52d63ed68f6fcaaf0b4faab0254c41171b9b311c/src/superqt/sliders/_labeled.py#L593-L596

but the value for the label minimum has already been reset by the time we get here - I can't figure out where it has been reset...